### ISSUE TITLE
What is required to build this repo

### ISSUE THREAD

**Issue by iplayfast** [rhasspy/piper#192]
I've added everytihing that I can see but still get this error.

chris@FORGE:~/ai/voice/piper$ make
mkdir -p build
cd build && cmake ../src/cpp -DCMAKE_BUILD_TYPE=Release && make
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/chris/ai/voice/piper/build
make[1]: Entering directory '/home/chris/ai/voice/piper/build'
make[2]: Entering directory '/home/chris/ai/voice/piper/build'
make[3]: Entering directory '/home/chris/ai/voice/piper/build'
make[3]: Leaving directory '/home/chris/ai/voice/piper/build'
make[3]: Entering directory '/home/chris/ai/voice/piper/build'
[ 33%] Building CXX object CMakeFiles/piper.dir/main.cpp.o
In file included from /home/chris/ai/voice/piper/src/cpp/main.cpp:29:
/home/chris/ai/voice/piper/src/cpp/piper.hpp:11:10: fatal error: onnxruntime_cxx_api.h: No such file or directory
   11 | #include <onnxruntime_cxx_api.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [CMakeFiles/piper.dir/build.make:76: CMakeFiles/piper.dir/main.cpp.o] Error 1
make[3]: Leaving directory '/home/chris/ai/voice/piper/build'
make[2]: *** [CMakeFiles/Makefile2:83: CMakeFiles/piper.dir/all] Error 2
make[2]: Leaving directory '/home/chris/ai/voice/piper/build'
make[1]: *** [Makefile:91: all] Error 2
make[1]: Leaving directory '/home/chris/ai/voice/piper/build'
make: *** [Makefile:9: piper] Error 2

**Comment by beqabeqa473** [comment]
hi.

build libPiperPhonemize first.

**Comment by iplayfast** [comment]
I guessed that meant piper-phonemize, (which couldn't work until I installed the forked espeak). 
Still didn't work. 
I think some work needs to be done on the documentation, to show how to do a complete install from scratch.

**Comment by iplayfast** [comment]
I went back to read the readme more closesly, and got it working, but it feels like it didn't build it, just downloaded it. I wish this repo had all the sub repo's in it and it was one build process.

**Comment by beqabeqa473** [comment]
everything is working if built using cmake

On 9/3/23, Chris Bruner ***@***.***> wrote:
> I went back to read the readme more closesly, and got it working, but it
> feels like it didn't build it, just downloaded it. I wish this repo had all
> the sub repo's in it and it was one build process.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/rhasspy/piper/issues/192#issuecomment-1704004333
> You are receiving this because you commented.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by synesthesiam** [comment]
I agree that the build process is a bit of a mess. I'm working on getting it to build fully in CMake on both Linux and Windows.

**Comment by Feanix-Fyre** [comment]
Anything we can do to help with the Windows compatibility development?

**Comment by synesthesiam** [comment]
Getting Github workflows for building, deploying, and testing piper-phonemize (https://github.com/rhasspy/piper-phonemize) and Piper itself on Linux, Windows, and Mac would help. But it's quite the tall order 😄

**Comment by synesthesiam** [comment]
Windows pre-release is available here: https://github.com/rhasspy/piper/releases/tag/2023.9.9-1
